x-pack/plugins/lists/server/routes/delete_list_route.ts

```diff
@@ -47,26 +49,31 @@ export const deleteListRoute = (router: ListsPluginRouter): void => {
 
         // ignoreReferences=true maintains pre-7.11 behavior of deleting value list without performing any additional checks
         if (!ignoreReferences) {
-          const referencedExceptionListItems = await exceptionLists.findValueListExceptionListItems(
-            {
-              page: 1,
-              perPage: 10000,
-              sortField: undefined,
-              sortOrder: undefined,
-              valueListId: id,
-            }
-          );
+          // Stream the results from the Point In Time (PIT) finder into this array
+          let referencedExceptionListItems: ExceptionListItemSchema[] = [];
+          const executeFunctionOnStream = (foundResponse: FoundExceptionListItemSchema): void => {
+            referencedExceptionListItems = [...referencedExceptionListItems, ...foundResponse.data];
+          };
 
-          if (referencedExceptionListItems?.data?.length) {
+          await exceptionLists.findValueListExceptionListItemsPointInTimeFinder({
+            executeFunctionOnStream,
+            maxSize: undefined, // NOTE: This is unbounded when it is "undefined"
+            perPage: 1_000,
```

Comment:
I know you put it in the PR description already, but maybe worth either linking to the PR or a quick note for our future selves on the 1000?

Reply:
Changed them to be:

```ts
perPage: 1_000, // See https://github.com/elastic/kibana/issues/93770 for choice of 1k
```

So there is some context on them.